### PR TITLE
Fix overlapping infantry and building decorations

### LINKED ISSUE #14736: No icon overlay for infantry when a hospital is captured in RA mod
The hospital does provide healing for infantry when captured, but there is no visual indication for that in RA mod. In TD mod there is a red cross icon overlay on each infantry. 

 Was it a deliberate decision to not include the icons for RA?

### PR DESCRIPTION
**Pull request by pchote**
This PR polishes our overlay decorations for infantry and structures.  The repair and powerdown icons are now shown side by side in RA/D2K/TS, and the heal/tiberium protection/veterancy icons now alternate on infantry where there is not enough space to do anything else.

This also adds the hospital heal indicator to RA, and changes it to blink and only display when the infantry is actively healing.  The hazmat indicator in TD is similarly only shown when the infantry is in a tiberium field.

Supersedes #14810.
Closes #14736.

### CODE REVIEW

**Comment by pchote** [comment]
Fixed the code style warnings.

**Comment by GraionDilach** [comment]
The yaml syntax looks really confusing on first sight.

File: mods/ra/rules/defaults.yaml
Comment (reaperrr) [inline mods/ra/rules/defaults.yaml]: This is really bound to confuse modders. In my opinion it would be more modder-friendly to introduce a `Blink` enum with `On`/ `Off` and map that to True/False internally.

Comment (pchote) [reply mods/ra/rules/defaults.yaml]: What if I rename the fields to `VisibilityInterval`, `VisibilityPattern`, `VisibilityPatterns` instead?

Comment (GraionDilach) [reply mods/ra/rules/defaults.yaml]: Wouldn't help, Blink sounds more reasonable here. I grasped the syntax now and tbh I don't think renaming this would help. It needs more detailed descriptions IMO.

**Comment by pchote** [comment]
Added blink On/Off enum.

**Comment by GraionDilach** [comment]
I would open an issue after this PR gets merged to refactor RepairableBuildings to utilize the stacked conditions for the repair effect as well.

**Review by GraionDilach — DISMISSED** [review]
:+1: otherwise.

**Comment by pchote** [comment]
Rebased.

**Review by Smittytron — DISMISSED** [review]
Looked good in testing

**Review by reaperrr — DISMISSED** [review]

**Comment by reaperrr** [comment]
Needs another rebase, LGTM after that.

**Comment by pchote** [comment]
Rebased.

File: OpenRA.Mods.Common/UpdateRules/UpdatePath.cs
Comment (pchote) [inline OpenRA.Mods.Common/UpdateRules/UpdatePath.cs]: I've deliberately added a comma here in the hopes that it will reduce the rebase conflicts going forward.

Comment (reaperrr) [reply OpenRA.Mods.Common/UpdateRules/UpdatePath.cs]: Makes sense, I'll try to keep that in mind for my PRs as well.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/506b83167b8da17dc0f94eeba345cdb718cd1876...992f30c8649588aadcbd84b4881dbbf9db4d53bc

**Comment by dragunoff** [comment]
I think that the blinking hospital indicator is a bit jarring. IMO it should be simply displayed, without blinking when infantry is healing. If it is to blink, then I'd suggest making it smoothly pulse by animating the opacity of the overlay. 

I'm not sure whether I should file a new issue for that, just my opinion here.

**Pull request merged**
